service/service.go

```diff
@@ -44,7 +45,8 @@ type Settings struct {
 	CollectorConf *confmap.Conf
 
 	// Receivers builder for receivers.
-	Receivers *receiver.Builder
+	ReceiversConfigs   map[component.ID]component.Config
+	ReceiversFactories map[component.Type]receiver.Factory
```

Comment:
We can't provide a `Builders` option anymore here, since that's private. I've gone with the "easy" way of providing the attributes directly, but we could do things differently?

Reply:
This new interface doesn't seem better than the builder fields.

Why can't we just add `CreateProfiles` to the existing builders instead?

cc @mx-psi

Reply:
Ok, after thinking more about it, it's probably good to remove the builders API as redundant anyway. I'm not sure that passing `ReceiversConfigs` and `ReceiversFactories` is the best option, but I don't have any other suggestions at this point

Reply:
An alternative would be to add `ReceiversConfigs` to the `service.Config` struct. @dmitryax what do you think about that option?

Reply:
As pointed out by @dmathieu, since 	`service.Config` is used by `otelcol.Config` we can't add the receivers config to `service.Config`.

Reply:
I think keeping in `Settings` is fine for now, we may want to refactor later but since `service` and `otelcol` Go APIs are out of scope for 1.0 this is not a pressing concern